actionview/lib/action_view/base.rb

```diff
@@ -237,11 +237,28 @@ def initialize(lookup_context, assigns, controller) # :nodoc:
       _prepare_context
     end
 
-    def _run(method, template, locals, buffer, add_to_stack: true, &block)
+    def _run(method, template, locals, buffer, add_to_stack: true, locals_disabled: false, explicit_locals: false, &block)
       _old_output_buffer, _old_virtual_path, _old_template = @output_buffer, @virtual_path, @current_template
       @current_template = template if add_to_stack
       @output_buffer = buffer
-      public_send(method, locals, buffer, &block)
+
+      if explicit_locals
+        if locals_disabled && locals.present?
+          raise ArgumentError, "#{template.short_identifier} does not accept locals."
+        else
+          begin
+            if locals_disabled
+              public_send(method, buffer, &block)
+            else
+              public_send(method, buffer, **locals, &block)
+            end
+          rescue ArgumentError => argument_error
+            raise ArgumentError, argument_error.message.gsub("keyword", "local")
```

Comment:
```suggestion
            raise ArgumentError, argument_error.message.sub("missing keyword:", "missing local:")
```

So that we don't also rewrite an argument called `foo_keyword` or something.